File: app/src/main/java/co/tpcreative/supersafe/common/preference/MySwitchPreference.java
package co.tpcreative.supersafe.common.preference;
import android.annotation.TargetApi;
import android.content.Context;
import android.os.Build;
import android.util.AttributeSet;
import android.view.View;
import androidx.appcompat.widget.SwitchCompat;
import androidx.preference.CheckBoxPreference;
import androidx.preference.PreferenceViewHolder;


public class MySwitchPreference extends CheckBoxPreference {

    private Context context;

    @TargetApi(Build.VERSION_CODES.LOLLIPOP)
    public MySwitchPreference(Context context, AttributeSet attrs, int defStyleAttr, int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);
        this.context = context;
        //init();
    }

    public MySwitchPreference(Context context) {
        super(context);
        this.context = context;
    }

    public MySwitchPreference(Context context, AttributeSet attrs) {
        super(context, attrs);
        this.context = context;
        //init();
    }

    public MySwitchPreference(Context context, AttributeSet attrs, int defStyle) {
        super(context, attrs, defStyle);
        this.context = context;
        //init();
    }

    @Override
    public void onBindViewHolder(PreferenceViewHolder view) {
        super.onBindViewHolder(view);
        SwitchCompat checkBox = (SwitchCompat) view.findViewById(android.R.id.checkbox);
        checkBox.setVisibility(View.VISIBLE);
    }

}

File: app/src/main/java/co/tpcreative/supersafe/common/preference/MySwitchPreference.kt
package co.tpcreative.supersafe.common.preference
import android.content.Context
import android.util.AttributeSet
import android.view.View
import androidx.appcompat.widget.SwitchCompat
import androidx.preference.CheckBoxPreference
import androidx.preference.PreferenceViewHolder
import co.tpcreative.supersafe.R

class MySwitchPreference @JvmOverloads constructor(
        context: Context,
        attrs: AttributeSet,
        defStyleAttr: Int = 0
) : CheckBoxPreference(context, attrs, defStyleAttr) {
    init {
        widgetLayoutResource = R.layout.custom_preferences_item
    }
    override fun onBindViewHolder(holder: PreferenceViewHolder) {
        super.onBindViewHolder(holder)
        with(holder.itemView) {
            val checkbox = this.findViewById(android.R.id.checkbox) as SwitchCompat
            checkbox.visibility = View.VISIBLE
        }
    }
}